### ISSUE TITLE
[Question][Documentation] Timing offsets

### ISSUE THREAD

**Issue by anisghaoui** [tyrex-team/senslogs#7]
Hello,
I was toying around with an IMU + camera record and I noticed that there might be a clock offset from SensLogs. I wanted to check this with you if possible.
The file `record.properties`:
```
[Device]
Manufacturer = OnePlus
Model = ONEPLUS A6010
OSVersion = 30
ID = 06bbcc95389bba64

[Settings]
User = 
PositionOrientation = 
Comment = 

[Time]
StartTime = 1647873406.975
EndTime = 1647873413.201
BootTime = 1644654255.779
MonotonicAtStart = 1697313.897

[Sensors]
List = lsm6ds3c Accelerometer Non-wakeup, ak0991x Magnetometer Non-wakeup, lsm6ds3c Gyroscope Non-wakeup, Camera

[Camera]
VideoOffset = 0.5530002117156982
```
To my understanding, the whole acquisition is  done via the same system clock but there is a trigger delay from the camera represented   by `VideoOffset` (in seconds?).
if so, this means the `*.txt` files' first column is the clock time relative to `StartTime`.

Did I get this right? or is there something else I need to take into account?
Thank you!

**Comment by ThibaudM** [comment]
For every recorded txt file, there are two timestamps. 
The first is the timestamp when the sensor value is received by the developer, and caculated using `System.currentTimeMillis() / 1e3d`.
The second is a timestamp returned by the sensor itself in the sensor's data.
I chose to return both timestamps in order to be able to simulate a "real-time system".

Now, about the camera. There is no specific data to know the precise time of the first frame of the video. So, I created the `VideoOffset` field which is calculated using `System.currentTimeMillis() / 1e3d` when the `CameraCaptureSession` is `onReady` (ref. https://github.com/tyrex-team/senslogs/blob/master/app/src/main/java/fr/inria/tyrex/senslogs/model/sensors/CameraRecorder.java#L261-L274). I do not know how this function is accurate, or accurate enough to do a Visual Inertial fusion, but hope that help.

To summarize, the VideoOffset field uses the same clock reference as the first column of the other file's data.

Hope this answers your question.

**Comment by anisghaoui** [comment]
Hi, 
This already answer my question. It means that `*.txt` files and video data are synchronised to `VideoOffset` precision (close to 0.5 micro sec, right?).
In my current experiments, there seems to be no synchronisation error for IMU-video fusion, relative to the the time precision my problem requires.
Thank you. This is clearer!

**Comment by ThibaudM** [comment]
> close to 0.5 micro sec, right?
yes
